src/Polly.Core/Simmy/Fault/FaultPipelineBuilderExtensions.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Polly.Simmy.Fault;
+
+namespace Polly.Simmy;
+
+/// <summary>
+/// Extension methods for adding outcome to a <see cref="ResiliencePipelineBuilder"/>.
+/// </summary>
+internal static partial class FaultPipelineBuilderExtensions
+{
+    /// <summary>
+    /// Adds a fault chaos strategy to the builder.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="injectionRate">The injection rate for a given execution, which the value should be between [0, 1] (inclusive).</param>
+    /// <param name="fault">The exception to inject.</param>
+    /// <returns>The builder instance with the retry strategy added.</returns>
+    public static ResiliencePipelineBuilder AddChaosFault(this ResiliencePipelineBuilder builder, double injectionRate, Exception fault)
```

Comment:
```suggestion
    public static TBuilder AddChaosFault<TBuilder>(this TBuilder builder, double injectionRate, Exception fault) where TBuilder: ResiliencePipelineBuilderBase
```

This will allow you to reduce a bunch od duplicate extensions.

Reply:
@vany0114 , unresolving as this is not yet addressed

Reply:
@martintmk  I did try your suggestion however it does not work, am I missing something? 

![image](https://github.com/App-vNext/Polly/assets/12032625/7500c0f3-f1c1-47f8-8711-80f5030999ed)

I also took a look at the Retry extensions and it is not implementing using that approach either, I guess it's because it does not work for generic strategies.

Reply:
It's still generic, but the generic constraint is on the method instead:
https://github.com/App-vNext/Polly/blob/a3ca9d070a8921e0cec54f8f43b26647ca4ec795/src/Polly.Core/ResilienceStrategy.cs#L23

Because you are just injection exceptions, there is no need to scope the generic to a single constrained generic parameter.

Reply:
Gotcha, I'll take a look next weekend then, I tried changing the class signature and it broke the unit tests.

Reply:
Done